src/Search/SearchResult.php

```diff
@@ -84,16 +94,53 @@ public function __construct(array $body)
      *
      * @return SearchResult
      */
```

Comment:
Is this not outdated?
```
/**
     * Return a new {@see SearchResult} instance with the hits filtered using `array_filter($this->hits, $callback, ARRAY_FILTER_USE_BOTH)`.
     *
     * The $callback receives both the current hit and the key, in that order.
     *
     * The method DOES not trigger a new search.
     *
     * @return SearchResult
     */
```

As per [Wikipedia](https://en.wikipedia.org/wiki/Fluent_interface#PHP): 
> In PHP, one can return the **current object** by using the `$this` special variable which represent the instance. Hence return $this; will make the method return the instance. The example below defines a class Employee and three methods to set its name, surname and salary. Each return the instance of the Employee class allowing to chain methods.


> Return a new {@see SearchResult} instance with the hits filtered using...

Should it not be `Return the current instance updated with the hits filtered using...

( I cannot suggest changed as it was not modified in this PR )

>      * The $callback receives both the current hit and the key, in that order.

Only some of the options require a callback.

I think unless I understand badly in which case sorry, it was not updated with the change from filter to applyOptions.

Reply:
Oh! Thanks! I forgot to update the comment!